dpcpp/stop/residual_norm_kernels.dp.cpp

```diff
@@ -52,7 +56,66 @@ namespace dpcpp {
 namespace residual_norm {
 
 
-constexpr int default_group_size = 512;
+constexpr int default_block_size = 256;
+
+
+template <typename ValueType>
+void residual_norm_kernel(size_type num_cols, ValueType rel_residual_goal,
```

Comment:
maybe implement it with sycl directly not porting

Reply:
I agree here, this would be a pretty straightforward kernel.